go/mysql/decimal/decimal.go

```diff
@@ -579,6 +579,14 @@ func (d Decimal) Exponent() int32 {
 	return d.exp
 }
 
+func (d Decimal) Size() int32 {
+	digitsCount := int32(len(d.value.String()))
+	if d.value.Sign() == -1 {
+		digitsCount--
+	}
+	return max(digitsCount, -d.exp)
+}
+
```

Comment:
This is just counting digits in the byte representation of the decimal, so there's no point on round-tripping it. As I've just suggested, a helper that operates directly on the original bytes will be simpler and more efficient.

Reply:
Made the change 💕!